conmon-rs/server/src/rpc.rs

```diff
@@ -79,10 +79,10 @@ fn pidfile_from_params(params: &conmon::CreateContainerParams) -> capnp::Result<
 }
 
 fn path_vec_from_text_list(tl: capnp::text_list::Reader) -> Result<Vec<PathBuf>, capnp::Error> {
-    let mut v: Vec<PathBuf> = vec![];
-    for t in tl {
-        let t_str = t?.to_string();
-        v.push(PathBuf::from(t_str));
-    }
-    Ok(v)
+    Ok(tl
+        .iter()
+        .collect::<Result<Vec<_>, _>>()?
+        .iter()
+        .map(PathBuf::from)
+        .collect::<Vec<_>>())
```

Comment:
But now we're allocating two `Vec<>`.  It's actually much simpler than that, see:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6a4412865d6c34aa039422c20e120830

Basically just `tl.into_iter().map(|r| r.map(PathBuf::from)).collect()` should work.

Reply:
Even better, thanks!